src/tools/PICS-generator/PICSGenerator.py

```diff
@@ -0,0 +1,429 @@
+            # console.print(f"Support: {supportElement.text}")
+            supportElement.text = "true"
+
+            # Since usage PICS (server or client) is not a list, we can break out when a match is found,
+            # no reason to keep iterating through the elements.
+            break
+
+    # Feature PICS
+    # console.print(featurePicsList)
+    featureNode = root.find("./clusterSide[@type='Server']/features")
+    for picsItem in featureNode:
+        itemNumberElement = picsItem.find('itemNumber')
+
+        console.print(f"Searching for {itemNumberElement.text}")
+
+        if f"{itemNumberElement.text}" in featurePicsList:
+            console.print("Found feature PICS value in XML template ✅")
+            supportElement = picsItem.find('support')
+            supportElement.text = "true"
+
+    # Attributes PICS
+    # TODO: Only check if list is not empty
+    # console.print(attributePicsList)
+    serverAttributesNode = root.find("./clusterSide[@type='Server']/attributes")
+    for picsItem in serverAttributesNode:
+        itemNumberElement = picsItem.find('itemNumber')
+
+        console.print(f"Searching for {itemNumberElement.text}")
+
+        if f"{itemNumberElement.text}" in attributePicsList:
+            console.print("Found attribute PICS value in XML template ✅")
+            supportElement = picsItem.find('support')
+            supportElement.text = "true"
+
+    # AcceptedCommandList PICS
+    # TODO: Only check if list is not empty
+    # console.print(acceptedCommandPicsList)
+    serverCommandsReceivedNode = root.find("./clusterSide[@type='Server']/commandsReceived")
+    for picsItem in serverCommandsReceivedNode:
+        itemNumberElement = picsItem.find('itemNumber')
+
+        console.print(f"Searching for {itemNumberElement.text}")
+
+        if f"{itemNumberElement.text}" in acceptedCommandPicsList:
+            console.print("Found acceptedCommand PICS value in XML template ✅")
+            supportElement = picsItem.find('support')
+            supportElement.text = "true"
+
+    # GeneratedCommandList PICS
+    # console.print(generatedCommandPicsList)
+    # TODO: Only check if list is not empty
+    serverCommandsGeneratedNode = root.find("./clusterSide[@type='Server']/commandsGenerated")
+    for picsItem in serverCommandsGeneratedNode:
+        itemNumberElement = picsItem.find('itemNumber')
+
+        console.print(f"Searching for {itemNumberElement.text}")
+
+        if f"{itemNumberElement.text}" in generatedCommandPicsList:
+            console.print("Found generatedCommand PICS value in XML template ✅")
+            supportElement = picsItem.find('support')
+            supportElement.text = "true"
+
+    # Event PICS (Work in progress)
+    # The ability to set event PICS is fairly limited, due to EventList not being supported,
+    # as well as no way to check for event support in the current Matter SDK.
+
+    # This implementation marks an event as supported if:
```

Comment:
As a follow up - I pulled out the python spec XML parsing into a separate function, and it has a list of all the events with their conformances. That might be able to be use here. Actually, it might be able to be used above too - ie, maybe we can unify these two as they develop. Feel free to just resolve this comment, btw - nothing to do here on this PR, but just wanted to mention it as a refinement

Reply:
That could be used as well. Currently it simply checks the conformance marked in the PICS XML file. If it is mandatory, either unconditionally or by a feature (which is supported), it is marked as supported. So not sure if there is much added value by getting this from the DM XML 🤷🏻‍♂️ 

The best improvement here would be to get the EventList sorted. 

I tried to dig into if there was anyway of poking at the events to figure out if it is supported or not, but I was not (yet) been able to find any solid way of doing this unfortunately.